enterprise/models.py

```diff
@@ -2733,13 +2733,38 @@ def __repr__(self):
         """
         return self.__str__()
 
+    @classmethod
+    def validate_compression(cls, enable_compression, data_type, delivery_method):
+        """Check enable_compression flag is set as expected
+
+        Args:
+            enable_compression (bool): file copression flag
+            data_type (string): report type
+            delivery_method (string): delivery method for sending files
+
+        Returns:
+           Validation Error
+        """
+
+        if not enable_compression and (
```

Comment:
can we simplify the condition here? I think following will be more readable.
```
if data_type in cls.ALLOWED_NON_COMPRESSION_DATA_TYPES or delivery_method == cls.DELIVERY_METHOD_SFTP:
    # Compression not needed and the can be disabled by the user.
    return
elif not enable_compression:
    allowed_data_types = ", ".join(cls.ALLOWED_NON_COMPRESSION_DATA_TYPES)
    error_message = (
        f'Compression can only be disabled for the following data types: {allowed_data_types} and '
         f'delivery method: {cls.DELIVERY_METHOD_SFTP}'
    )
    return {'enable_compression': error_message}
```

Reply:
This looks good to me but there needs some changes in logic.
```
if data_type in cls.ALLOWED_NON_COMPRESSION_DATA_TYPES and delivery_method == cls.DELIVERY_METHOD_SFTP:
    # Compression not needed and the can be disabled by the user.
    return {}
elif not enable_compression:
    allowed_data_types = ", ".join(cls.ALLOWED_NON_COMPRESSION_DATA_TYPES)
    error_message = (
        f'Compression can only be disabled for the following data types: {allowed_data_types} and '
         f'delivery method: {cls.DELIVERY_METHOD_SFTP}'
    )
    return {'enable_compression': error_message}
return {}
```